src/btree/bt_compact.c

```diff
@@ -103,21 +103,44 @@ __compact_rewrite_lock(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
 	return (ret);
 }
 
+/*
+ * __compact_progress --
+ *     Output a compact progress message.
+ */
+static void
+__compact_progress(WT_SESSION_IMPL *session, uint64_t time_diff)
+{
+	WT_BM *bm;
+
+	bm = S2BT(session)->bm;
+	__wt_verbose(session,
+	    WT_VERB_COMPACT_PROGRESS, "Compact running"
+	    " for %" PRIu64 " seconds; reviewed %"
+	    PRIu64 " pages, skipped %" PRIu64 " pages,"
+	    " wrote %" PRIu64 " pages", time_diff,
+	    bm->block->compact_pages_reviewed,
+	    bm->block->compact_pages_skipped,
+	    bm->block->compact_pages_written);
+}
+
 /*
  * __wt_compact --
  *	Compact a file.
  */
 int
 __wt_compact(WT_SESSION_IMPL *session)
 {
+	struct timespec now, last;
 	WT_BM *bm;
 	WT_DECL_RET;
 	WT_REF *ref;
+	uint64_t time_diff;
 	u_int i;
 	bool skip;
 
 	bm = S2BT(session)->bm;
 	ref = NULL;
+	__wt_epoch(session, &last);
```

Comment:
Do we want to consider `wt_clock` here? Although we only print the progress messages once per second, we may be coming through here more frequently.

Reply:
Yes, it's worth using `__wt_clock`.